### PR TITLE
2024: Document cargo changes

### PR DESCRIPTION
**Pull request by ehuss**
This adds some documentation for some of the cargo changes.

cc https://github.com/rust-lang/rust/issues/123754
cc https://github.com/rust-lang/rust/issues/123750

cc @rust-lang/cargo

### CODE REVIEW

File: src/rust-2024/cargo-table-key-names.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+🚧 The 2024 Edition has not yet been released and hence this section is still "under construction".
+
+## Summary
+
+- Several table and key names in `Cargo.toml` have been removed where there was previously two ways to specify it.
+    - Removed `[project]`, use `[package]` instead.
+    - Removed `default_features`, use `default-features` instead.
+    - Removed `crate_type`, use `crate-type` instead.
+    - Removed `dev_dependencies`, use `dev-dependencies` instead.
+    - Removed `build_dependencies`, use `build-dependencies` instead.
```
Comment (epage) [inline src/rust-2024/cargo-table-key-names.md:12]:
```suggestion
    - Removed `[project]`, use `[package]` instead.
    - Removed `default_features`, use `default-features` instead.
    - Removed `crate_type`, use `crate-type` instead.
    - Removed `proc_macro`, use `proc-macro` instead.
    - Removed `dev_dependencies`, use `dev-dependencies` instead.
    - Removed `build_dependencies`, use `build-dependencies` instead.
```

Comment (epage) [reply src/rust-2024/cargo-table-key-names.md:12]: (proc_macro)

**Review by epage — APPROVED** [review]
Assuming the proc_macro part is fixed

**Review by weihanglo — APPROVED** [review]

File: src/rust-2024/cargo-table-key-names.md (hunk 1 above)
Comment (ehuss) [reply src/rust-2024/cargo-table-key-names.md]: Thanks! Not sure how I missed it since it was in my notes.

**Review by traviscross — APPROVED** [review]

**Review by Findrown — APPROVED** [review]

**Review by Findrown — COMMENTED** [review]
Done

**Pull request merged**